docs: clarify that the options of `ng lint` in AIO are of the deprecated builder (#19649)
angular/angular-cli: packages/angular/cli/commands/lint-long.md

Takes the name of the project, as specified in the  `projects` section of the `angular.json` workspace configuration file.
When a project name is not supplied, it will execute for all projects.

The default linting tool is [TSLint](https://palantir.github.io/tslint/), and the default configuration is specified in the project's `tslint.json` file.

> **Note**: TSLint's support is discontinued and we're deprecating its support in Angular CLI and the options below are for the deprecated builder.

Reviewer: ```suggestion
**Note**: TSLint has been discontinued and support has been deprecated in the Angular CLI. The options shown below are for the deprecated TSLint builder.
```
